pkg/cache/cache.go

```diff
@@ -299,9 +299,20 @@ func New(defaultAudience, prefix string, defaultRegionalSTS bool, defaultTokenEx
 		defaultTokenExpiration: defaultTokenExpiration,
 		hasSynced:              hasSynced,
 		webhookUsage:           webhookUsage,
-		notificationHandlers:   map[string]chan struct{}{},
+		notifications:          newNotifications(saFetchRequests),
 	}
 
+	go func() {
+		for req := range saFetchRequests {
```

Comment:
This would mean we are making only one request at a time to apiserver right?

Reply:
Wrapped this in a goroutine - thanks for the catch 💯